discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -515,3 +529,72 @@ def wait_for_cluster_to_be_in_installing_status(self):
+        # Reboot required nodes into ISO
+        cluster.reboot_required_nodes_into_iso_after_reset(nodes=nodes)
+        # Wait for hosts to be rediscovered
+        cluster.wait_until_hosts_are_discovered()
+        cluster.wait_for_ready_to_install()
+
+    def verify_no_logs_uploaded(self, cluster, path):
+        with pytest.raises(ApiException) as ex:
+            cluster.download_installation_logs(path)
+        assert "No log files" in str(ex.value)
+
+    def verify_logs_uploaded(self, path, expected_min_log_num, installation_success):
+
+        # verify that logs were collected from all expected sources
+        assert (os.path.exists(path))
+        dir_path = path.split(".")[0]
+        try:
+            tar = tarfile.open(path)
+            logging.info(f'downloaded logs: {tar.getnames()}')
+            assert len(tar.getnames()) >= expected_min_log_num
+            tar.extractall(dir_path)
+            for gz in os.listdir(dir_path):
+                if "bootstrap" in gz:
+                    self._verify_node_logs_uploaded(dir_path, gz)
+                    self._verify_bootstrap_logs_uploaded(dir_path, gz, installation_success)
+                elif "master" in gz or "worker" in gz:
+                    self._verify_node_logs_uploaded(dir_path, gz)
+            tar.close()
+        finally:
+            # clean up
+            path = os.path.abspath(path)
+            os.remove(path)
+            shutil.rmtree(dir_path)
+
+    def _verify_node_logs_uploaded(self, dir_path, file_path):
+        gz = tarfile.open(os.path.join(dir_path, file_path))
+        logs = gz.getnames()
+        assert any("agent.logs" in s for s in logs)
+        assert any("installer.logs" in s for s in logs)
+        assert any("mount.logs" in s for s in logs)
+        gz.close()
+
+    def _verify_bootstrap_logs_uploaded(self, dir_path, file_path, installation_success):
+        gz = tarfile.open(os.path.join(dir_path, file_path))
+        logs = gz.getnames()
+        assert any("bootkube.logs" in s for s in logs)
+        if not installation_success:
+            assert any("dmesg.logs" in s for s in logs)
+            assert any("log-bundle" in s for s in logs)
+            # test that installer-gather gathered logs from all masters
+            lb_path = [s for s in logs if "log-bundle" in s][0]
+            gz.extract(lb_path, dir_path)
+            lb = tarfile.open(os.path.join(dir_path, lb_path))
+            lb.extractall(dir_path)
+            cp_path = [s for s in lb.getnames() if "control-plane" in s][0]
+            assert len(os.listdir(os.path.join(dir_path, cp_path))) == env_variables["num_masters"]-1
+            lb.close()
+        gz.close()
+
+    def verify_logs_are_current(self, started_cluster_install_at, logs_collected_at):
+        for collected_at in logs_collected_at:
+            # if host timestamp is set at all- check that the timestamp is from the last installation
+            if collected_at > time.time() - 86400000:
```

Comment:
Please make a constant value, what does `86400000` stands for?

Reply:
Same, this is existing code, we can change it in a different PR.